### DISCUSSION TITLE
Material appearance of .glb file

### DISCUSSION THREAD

**Discussion by watsonovich** [google/model-viewer#2404] [Q&A] [answered]
This may not be the right place to post this question, but I really don't know where else to start.  I am hoping someone can simply point me in the right troubleshooting direction.  My project does not use model-viewer, but I've used modelviewer.dev/editor to help me visualize my progress in authoring .glb files.  My project is structured based on guides on discoverthreejs.com.

When I load my model into modelviewer.dev/editor it looks like this: 
![image](https://user-images.githubusercontent.com/15159132/118848772-0cdb3180-b89d-11eb-96e4-591dbe88c339.png)

When I load the same assets into my three.js scene the metallic textures are simply a glossy black, like this:
![image](https://user-images.githubusercontent.com/15159132/118848938-3ac07600-b89d-11eb-9858-8bea9ad63c92.png)

Can anyone offer very basic troubleshooting tips?    Does model-viewer go deeper into my .glb file to specifically access these textures?  Basically, do I need to start with lighting, shaders, metalness, or am I just not applying textures that are in the .glb file? 
 I have no idea how to share coding examples, but here is my project if that is helpful: 
https://watsonovich.github.io/glbLoader/
https://github.com/watsonovich/glbLoader

I just noticed in Chrome dev tools like 180ish warnings about "texture has to be resized."  Maybe that's where I need some direction.  Why would that be necessary?

**Comment by shamsail** [comment, marked as answer]
Hi @watsonovich  , I can asume what's happening there and nothing is wrong with your textures. In modelviewer/editor and model viewer library , a default environment map is being used that reflects on metalic things or textures. And this environment map for metalic textures is necessary for a good look. 
You need to load environment map in your threejs scene. May be you can use

```
import { PMREMGenerator, UnsignedByteType } from "three";

const pmremGenerator = new PMREMGenerator(renderer);
        pmremGenerator.compileEquirectangularShader();
        new RGBELoader()
          .setDataType(UnsignedByteType)
          .load("path/to/enmap.hdr", function (texture) {
            var envMap = pmremGenerator.fromEquirectangular(texture).texture;

            scene.environment = envMap;

            texture.dispose();
            pmremGenerator.dispose();
          });
```
         It will resolve your black issue regarding metalic textures

**Reply by watsonovich** [reply]
That was exactly what I needed.  Thank you!  I'm so glad I asked... I'm not a developer, it would have likely taken me weeks to figure this out.  I still like the look from the modelviewer much better, but I'll study the PMREM and related things to see what settings I can play around with.  Again, Thanks!
![image](https://user-images.githubusercontent.com/15159132/119192268-d7c21100-ba4d-11eb-9862-8c932bf9912e.png)
